app/lib/collator_mint.py

```diff
@@ -107,13 +108,14 @@ def collator_set_keys(node_name, para_id, ss58_format):
         node_client = get_node_client(node_name)
         collator_account_funds = get_account_funds(node_client.url, collator_account_address)
         # 3. If insufficient, add funds with teleport
-        if collator_account_funds < 0.5 * 10 ** 12:
+        token_decimals = get_chain_properties(node_client)['tokenDecimals']
+        if collator_account_funds < 0.5 * 10 ** token_decimals:
             relay_chain_client = get_relay_chain_client()
             sudo_keypair = Keypair.create_from_seed(network_sudo_seed())
             log.info(f"Funding {collator_account_address}[ss58format={ss58_format}](funds={collator_account_funds}) via Teleport from relay-chain")
             # Get corresponding collator account address on the relay-chain (with the relay-chain ss58 format)
             relay_chain_collator_account = get_derived_collator_keypair(node_name, get_network_ss58_format()).ss58_address
-            teleport_result = teleport_funds(relay_chain_client, sudo_keypair, para_id, [relay_chain_collator_account], 1 * 10 ** 12)
+            teleport_result = teleport_funds(relay_chain_client, sudo_keypair, para_id, [relay_chain_collator_account], 1 * 10 ** token_decimals)
```

Comment:
```suggestion
            teleport_result = teleport_funds(relay_chain_client, sudo_keypair, para_id, [relay_chain_collator_account], 1)
```
`teleport_funds` function contains `10 ** token_decimals`